source/MaterialXGenShader/OpenColorIOManagementSystem.cpp

```diff
@@ -0,0 +1,198 @@
+//
+// Copyright Contributors to the MaterialX Project
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#ifdef MATERIALX_BUILD_OCIO
+
+#include "MaterialXCore/Library.h"
+
+#include <MaterialXGenShader/OpenColorIOManagementSystem.h>
+
+#include <MaterialXCore/Definition.h>
+#include <MaterialXGenShader/ColorManagementSystem.h>
+#include <MaterialXGenShader/ShaderGenerator.h>
+#include <MaterialXGenShader/Nodes/OpenColorIONode.h>
+
+#include <OpenColorIO/OpenColorIO.h>
+#include <OpenColorIO/OpenColorTypes.h>
+
+#include <exception>
+#include <stdexcept>
+#include <map>
+#include <string>
+
+MATERIALX_NAMESPACE_BEGIN
+
+const string OpenColorIOManagementSystem::IMPL_PREFIX = "IMPL_MXOCIO_";
+const string OpenColorIOManagementSystem::ND_PREFIX = "ND_MXOCIO_";
+
+namespace
+{
+
+const string CMS_NAME = "OpenColorIO";
+
+// Remap from legacy color space names to their ACES 1.3 equivalents.
+const std::map<string, string> COLOR_SPACE_REMAP = {
+    { "gamma18", "Gamma 1.8 Rec.709 - Texture" },
+    { "gamma22", "Gamma 2.2 Rec.709 - Texture" },
+    { "gamma24", "Gamma 2.4 Rec.709 - Texture" },
+    // TODO: Add support for adobergb and lin_adobergb
+};
+
+} // anonymous namespace
+
+//
+// OpenColorIOManagementSystem methods
+//
+
+OpenColorIOManagementSystemPtr OpenColorIOManagementSystem::create(const OCIO::ConstConfigRcPtr& config, const string& target)
+{
+    if (target != "genglsl" && target != "genmsl" && target != "genosl")
```

Comment:
targets can also be inherited from each other, and I believe that maybe OpenUSD's Storm uses this concept.  Instead of just having a fix list of base target names, we may perhaps need to respect inherited targets as well?